core/src/main/java/org/apache/carbondata/core/util/BlockletDataMapUtil.java

```diff
@@ -257,13 +257,24 @@ public static boolean isCacheLevelBlock(CarbonTable carbonTable) {
    * name but with different dataType.
    */
   public static boolean isSameColumnAndDifferentDatatypeInSchema(
-      List<ColumnSchema> indexFileColumnList, List<ColumnSchema> tableColumnList) {
+      List<ColumnSchema> indexFileColumnList, List<ColumnSchema> tableColumnList)
+      throws IOException {
     for (int i = 0; i < tableColumnList.size(); i++) {
       for (int j = 0; j < indexFileColumnList.size(); j++) {
         if (indexFileColumnList.get(j).getColumnName()
             .equalsIgnoreCase(tableColumnList.get(i).getColumnName()) && !indexFileColumnList.get(j)
             .getDataType().getName()
             .equalsIgnoreCase(tableColumnList.get(i).getDataType().getName())) {
+          if ("varchar".equalsIgnoreCase(indexFileColumnList.get(j).getDataType().getName()) &&
+              "string".equalsIgnoreCase(tableColumnList.get(i).getDataType().getName())) {
+            throw new IOException("Datatype of the Column "
```

Comment:
why you are throwing IO exception when it is not related IO ??

Reply:
optimized